### DISCUSSION TITLE
Detecting Different Form Factors

### DISCUSSION THREAD

**Discussion by uditadhikari117** [twilio-labs/paste#2508] [General] [answered]
Do we have some utility that can detect when the browser has be resized to a phone size?  I have temporary solution that I can share on how I handle this but wanted to check with paste. Following are two different images that are displayed in different positions in JSX (not shown here) and with different sizes depending on the form factor of the browser (desktop or phone).

![Email_Phone](https://user-images.githubusercontent.com/97563897/176704716-cc52d67e-770b-4039-8b95-9f8b7c1d2922.png)
![Email](https://user-images.githubusercontent.com/97563897/176704719-86100730-7d85-4c93-a4d4-0086b092e38b.png)

**Comment by gloriliale** [comment, marked as answer]
Hi @uditadhikari117! I'd like to know a little more about what you're asking. If the question is referring to a utility to sense browser size, then no, we do not have something like that specific to Paste. Is your question specifically about images? I noticed you're not using a responsive image, but wasn't sure if that was a part of your question. Let me know!

**Reply by gloriliale** [reply]
I also wanted to add: the #eng-guild-frontend channel would be a good place to talk with other devs about this kind of topic!

**Reply by uditadhikari117** [reply]
Hi @gloriliale ,
Thanks for looking into it. We are looking for a way to detect broswer width in mobile or desktop view. We do have a solution in a custom hook here :
https://github.com/twilio-internal/console/blob/GROW-3838/src/applications/AccountVerification/Hooks/useFormFactor.ts
We will later optimize this to work inside a provider. Happy to work with you guys if you want to absorb this and make it available for everyone.

**Reply by TheSisb** [reply]
Hi @uditadhikari117,

Thanks for sharing your hook, it's helpful to see!  I just noticed we had [built something similar](https://github.com/twilio-labs/paste/blob/main/packages/paste-website/src/hooks/useWindowSize.tsx) for the Paste documentation website but didn't expose it publicly. We may consider making this part of our package.  We appreciate your input!
